[jit][edge] Load interface methods to corresponding ClassTypes.

zhxchen17 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #67483
* #65972
* __->__ #65971
* #65970
* #65969





We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

zhxchen17 committed:
[jit][edge] Load interface methods to corresponding ClassTypes.



Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."


Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

qihqi reviewed (commented):
I feel like this change should come with a corresponding change in export_module.cpp.
(edit: nvm; found this https://github.com/pytorch/pytorch/pull/65966/files)

Currently we only serialize functions of methods attached to the module being serialized.
So this registry will put all those methods into the module's ClassType.

It's unclear that we want to write methods belonging to a different ClassType in the same place, do we?

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

iseeyuan reviewed (approved):
LGTM

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]

zhxchen17 committed:
Update on "[jit][edge] Load interface methods to corresponding ClassTypes."






We should be able to load methods into their ClassTypes. Right now mobile runtime only loads data member to ClassTypes but not for methods. To support interface call, we inject methods into ClassTypes when the methods are loaded.

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146/)

Differential Revision: [D31326146](https://our.internmc.facebook.com/intern/diff/D31326146)

[ghstack-poisoned]
